pycbc/derived_parameters.py

```diff
@@ -0,0 +1,462 @@
+# Copyright (C) 2017  Collin Capano
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides a library of functions that calculate waveform parameters
+from other parameters. All exposed functions in this module's namespace return
+one parameter given a set of inputs.
+"""
+
+import copy
+import numpy
+import lal
+from pycbc import coordinates
+from pycbc.detector import Detector
+
+#
+# =============================================================================
+#
+#                           Helper functions
+#
+# =============================================================================
+#
+def _ensurearray(arg):
```

Comment:
@ahnitz I think Collin has done what you're asking. He's just doing all the internal math as arrays. Look at the ``_formatreturn`` function.

Testing it out, seems like it works:
```
In [2]: derived_parameters.mass2_from_mchirp_q(1.4, 0.25)
Out[2]: 0.8407871037870017
```